src/react/RegisterDevicePage.tsx

```diff
@@ -264,28 +320,31 @@ const RegisterDevicePage = ({ astarte, history }) => {
 
   const generateRandomUUID = useCallback(() => {
     const newUUID = uuidv4().replace(/-/g, '');
-    const bytes = newUUID.match(/.{2}/g).map((b) => parseInt(b, 16));
+    const bytes = (newUUID.match(/.{2}/g) as RegExpMatchArray).map((b) => parseInt(b, 16));
     const newDeviceID = byteArrayToUrlSafeBase64(bytes);
-
     setDeviceId(newDeviceID);
   }, []);
 
-  const registerDevice = (e) => {
+  const registerDevice = (e: React.FormEvent<HTMLElement>) => {
     e.preventDefault();
-
-    const params = { deviceId };
-    if (sendIntrospection) {
-      params.introspection = introspectionInterfaces;
-    }
-
+    const deviceIntrospection: { [interfaceName: string]: AstarteInterfaceDescriptor } = {};
+    Array.from(introspectionInterfaces).forEach(([key, interfaceDescriptor]) => {
+      deviceIntrospection[key] = {
```

Comment:
you can copy the `interfaceDescriptor` directly here

Reply:
You're right. Actually I simplified even more since a Map can be directly converted into an indexed object with `Object.fromEntries`